src/components/views/pips/WidgetPip.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { FC, useCallback, useMemo, useRef } from "react";
+import { Room, RoomEvent } from "matrix-js-sdk/src/models/room";
+
+import PersistentApp from "../elements/PersistentApp";
+import defaultDispatcher from "../../../dispatcher/dispatcher";
+import { ViewRoomPayload } from "../../../dispatcher/payloads/ViewRoomPayload";
+import { Action } from "../../../dispatcher/actions";
+import { useCallForWidget } from "../../../hooks/useCall";
+import WidgetStore from "../../../stores/WidgetStore";
+import { Container, WidgetLayoutStore } from "../../../stores/widgets/WidgetLayoutStore";
+import { useTypedEventEmitterState } from "../../../hooks/useEventEmitter";
+import Toolbar from "../../../accessibility/Toolbar";
+import { RovingAccessibleButton, RovingAccessibleTooltipButton } from "../../../accessibility/RovingTabIndex";
+import PictureInPictureDragger, { CreatePipChildren } from "../../structures/PictureInPictureDragger";
+import { Icon as BackIcon } from "../../../../res/img/element-icons/back.svg";
+import { Icon as HangupIcon } from "../../../../res/img/element-icons/call/hangup.svg";
+import { _t } from "../../../languageHandler";
+import { WidgetType } from "../../../widgets/WidgetType";
+import { WidgetMessagingStore } from "../../../stores/widgets/WidgetMessagingStore";
+import WidgetUtils from "../../../utils/WidgetUtils";
+import { ElementWidgetActions } from "../../../stores/widgets/ElementWidgetActions";
+import { Alignment } from "../elements/Tooltip";
+
+interface Props {
+    widgetId: string;
+    room: Room;
+    viewingRoom: boolean;
+}
+
+/**
+ * A picture-in-picture view for a widget.
+ */
+export const WidgetPip: FC<Props> = ({ widgetId, room, viewingRoom }) => {
```

Comment:
This component looks very tightly coupled with calls. Should its name be a bit more specific? As at first glance it looks rather generic, but can't be re-used.

Reply:
I've left the name alone since this component really is capable of displaying any persistent widget, even ones that aren't for calls, but I have added a mention of its call-specific behavior in the comment now.